synapse/axon.py

```diff
@@ -1015,6 +1019,74 @@ async def jsonlines(self, sha256):
                 raise s_exc.BadJsonText(mesg=f'Bad json line encountered while processing {sha256}, ({e})',
                                         sha256=sha256) from None
 
+    async def postfiles(self, fields, url, params=None, headers=None, method='POST', ssl=True, timeout=None):
+        '''
+        Send files from the axon as fields in a multipart/form-data HTTP request.
+        '''
+        proxyurl = self.conf.get('http:proxy')
+        cadir = self.conf.get('tls:ca:dir')
+
+        connector = None
+        if proxyurl is not None:
+            connector = aiohttp_socks.ProxyConnector.from_url(proxyurl)
+
+        if ssl is False:
+            pass
+        elif cadir:
+            ssl = s_common.getSslCtx(cadir)
+        else:
+            # default aiohttp behavior
+            ssl = None
+
+        atimeout = aiohttp.ClientTimeout(total=timeout)
+
+        async with aiohttp.ClientSession(connector=connector, timeout=atimeout) as sess:
+
+            try:
+                data = aiohttp.FormData()
+                for field in fields:
+                    sha256 = field.get('sha256')
+                    if sha256:
+                        byts = []
+                        async for chunk in self.get(sha256):
+                            byts.append(chunk)
+                        valu = b''.join(byts)
```

Comment:
Can't do this if the file is huge, may need to use a SpooledTemporaryFile. Does the content option support a file-like object?

Reply:
I believe so, I'll give that a try

Reply:
Got this working but it does not play nicely with vcrpy.